socorro/unittest/external/boto/test_connection_context.py

```diff
@@ -53,40 +48,31 @@ class ConditionallyABadDeal(Exception):
 thing_as_str = json.dumps(a_thing)
 
 
-class ConnectionContextTestCase(socorro.unittest.testbase.TestCase):
-    def setup_mocked_s3_storage(
-        self,
-        executor=TransactionExecutor,
-        executor_for_gets=TransactionExecutor,
-        storage_class='BotoS3CrashStorage',
-        host='',
-        port=0,
-        resource_class=S3ConnectionContext,
-        **extra
-    ):
-        config = DotDict({
-            'resource_class': resource_class,
-            'logger': mock.Mock(),
-            'host': host,
-            'port': port,
-            'access_key': 'this is the access key',
-            'secret_access_key': 'secrets',
-            'bucket_name': 'silliness',
-            'keybuilder_class': KeyBuilderBase,
-            'prefix': 'dev',
-            'calling_format': mock.Mock()
-        })
-        config.update(extra)
-        s3_conn = resource_class(config)
-        s3_conn._connect_to_endpoint = mock.Mock()
-        s3_conn._mocked_connection = s3_conn._connect_to_endpoint.return_value
-        s3_conn._calling_format.return_value = mock.Mock()
-        s3_conn._CreateError = mock.Mock()
-        s3_conn.ResponseError = mock.Mock()
-        s3_conn._open = mock.MagicMock()
-
-        return s3_conn
+def setup_mocked_s3_storage(cls=S3ConnectionContext, **extra):
+    values_source = {
+        'bucket_name': 'silliness',
+        'prefix': 'dev',
+        'calling_format': mock.Mock()
+    }
+    values_source.update(extra)
+
+    config = get_config(
+        cls=cls,
+        values_source=values_source
+    )
+
+    s3_conn = cls(config)
+    s3_conn._connect_to_endpoint = mock.Mock()
+    s3_conn._mocked_connection = s3_conn._connect_to_endpoint.return_value
+    s3_conn._calling_format.return_value = mock.Mock()
+    s3_conn._CreateError = mock.Mock()
+    s3_conn.ResponseError = mock.Mock()
+    s3_conn._open = mock.MagicMock()
+
+    return s3_conn
 
+
+class TestConnectionContext:
```

Comment:
The rest of the changes in this file are mostly cosmetic:

* connection_source -> conn
* reducing some vertical stuff into single lines
* changing classes so they don't inherit from `unittest.TestCase`